packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -416,4 +441,29 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
     if (providers.length === 0) { return undefined; }
     return Array.from(providers);
   }
+
+  private configureTokenValidity(resource: CfnUserPoolClient, props: UserPoolClientProps) {
+    this.validateDurationInRangeOrUndefined('idTokenValidity', Duration.minutes(5), Duration.days(1), props.idTokenValidity);
+    this.validateDurationInRangeOrUndefined('accessTokenValidity', Duration.minutes(5), Duration.days(1), props.accessTokenValidity);
+    this.validateDurationInRangeOrUndefined('refreshTokenValidity', Duration.minutes(60), Duration.days(10 * 365), props.refreshTokenValidity);
+
```

Comment:
Also validate that id tokens and access tokens are within the refresh token limits. You should be able to use the same `validateDuration...` method.

Reply:
Good remark, I added the additional validation and created the test cases for it.